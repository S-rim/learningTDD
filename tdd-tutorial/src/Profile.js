import React from "react";

const Profile = ({ username, name }) => {
  return (
    <div>
      <strong>{username}</strong>
      <span>({name})</span>
    </div>
  );
};

export default Profile;
