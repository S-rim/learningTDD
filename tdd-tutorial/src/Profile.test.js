import React from "react";
import { render } from "@testing-library/react";
import Profile from "./Profile";

describe("<Profile />", () => {
  it("matches snapshot", () => {
    const utils = render(<Profile username="S-rim" name="신서림" />);
    expect(utils.container).toMatchSnapshot();
  });
  it("shows the props correctly", () => {
    const utils = render(<Profile username="S-rim" name="신서림" />);
    // 해당 텍스트 가지거나 정규식을 통과하는 엘리먼트가 있는지 확인
    utils.getByText("S-rim");
    utils.getByText("(신서림)");
    utils.getByText(/신/);
  });
});
