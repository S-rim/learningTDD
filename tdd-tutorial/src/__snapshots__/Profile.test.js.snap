// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshot 1`] = `
<div>
  <div>
    <strong>
      S-rim
    </strong>
    <span>
      (
      신서림
      )
    </span>
  </div>
</div>
`;
